.github/workflows/checks.yml

```diff
@@ -8,6 +8,8 @@ on:
 jobs:
   test:
     name: Test
+    permissions:
```

Comment:
We should also add `permissions: {}` on the top level, then, since default access levels for different scopes can be much broader: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token.